marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/step/saveStep.sjs

```diff
@@ -43,7 +43,16 @@ if (existingStep) {
 
   // For now, can assume the stepDefinitionName based on the type. Can add stepDefinitionType as a parameter once we need
   // more flexibility.
-  const stepDefinitionName = "mapping" === stepDefinitionType.toLowerCase() ? "entity-services-mapping" : "default-ingestion";
+  let stepDefinitionName;
+  if("mapping" === stepDefinitionType){
+    stepDefinitionName = "entity-services-mapping";
+  }
+  else if("ingestion" === stepDefinitionType){
+    stepDefinitionName = "default-ingestion";
+  }
+  else {
+    stepDefinitionName = stepProperties.stepDefinitionName;
```

Comment:
I think an exception should be thrown here if stepDefinitionName is null.

Reply:
In custom.sjs "stepDefinitionName" is a mandatory property and will throw an error if it is absent. Shouldn't that be sufficient ?

Reply:
Good point, that works.